internal/corazawaf/transaction.go

```diff
@@ -794,28 +796,38 @@ func (tx *Transaction) WriteRequestBody(b []byte) (*types.Interruption, int, err
 
 	if tx.RequestBodyLimit == tx.requestBodyBuffer.length {
 		// tx.RequestBodyLimit will never be zero so if this happened, we have an
-		// interruption for sure.
-		if tx.WAF.RequestBodyLimitAction == types.RequestBodyLimitActionReject {
+		// interruption (that has been previously raised, but ignored by the connector) for sure.
+		if tx.WAF.RequestBodyLimitAction == types.BodyLimitActionReject {
 			return tx.interruption, 0, nil
 		}
 
-		if tx.WAF.RequestBodyLimitAction == types.RequestBodyLimitActionProcessPartial {
+		if tx.WAF.RequestBodyLimitAction == types.BodyLimitActionProcessPartial {
 			return nil, 0, nil
 		}
 	}
 
 	var (
 		writingBytes          = int64(len(b))
 		runProcessRequestBody = false
+		targetLen             int64
 	)
 
-	if tx.requestBodyBuffer.length+writingBytes >= tx.RequestBodyLimit {
-		if tx.WAF.RequestBodyLimitAction == types.RequestBodyLimitActionReject {
+	// Overflow check
+	if tx.requestBodyBuffer.length == math.MaxInt64 || tx.requestBodyBuffer.length >= (math.MaxInt64-writingBytes) {
+		// Overflow, buffer length will always be at most MaxInt
+		targetLen = math.MaxInt64
```

Comment:
I think this was my idea originally but I would just fail here (i.e. return an error). This isn't a realistic payload size and we are adding this to be programatic correct but we don't need to support this case. More so, buffer won't accept this kind of sizes as we experienced in real life so let's just fail.